packages/cozy-harvest-lib/src/components/Routes.jsx

```diff
@@ -10,17 +10,30 @@ import NewAccountModal from './NewAccountModal'
 import EditAccountModal from './EditAccountModal'
 import KonnectorSuccess from './KonnectorSuccess'
 
-const Routes = ({ konnectorRoot, konnector, location, onDismiss }) => (
+const Routes = ({ konnectorRoot, konnector, location, history, onDismiss }) => (
   <Modal dismissAction={onDismiss} mobileFullscreen size="small">
     <KonnectorAccounts konnector={konnector}>
       {accounts => (
         <>
           <Route
             path={`${konnectorRoot}/`}
             exact
-            render={() => (
-              <AccountsListModal konnector={konnector} accounts={accounts} />
-            )}
+            render={() => {
+              if (accounts.length === 0) {
+                history.push('./new')
+                return null
+              } else if (accounts.length === 1) {
+                history.push(`./accounts/${accounts[0].account._id}`)
+                return null
+              } else {
```

Comment:
Seriously? Is it working? No redirecting loop even without '/' in the first place?